src/lib/gcp_apis.py

```diff
@@ -11,32 +11,32 @@
 #     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #     See the License for the specific language governing permissions and
 #     limitations under the License.
+from aiohttp import ClientResponseError
 
 from lib.context import MetricsContext
 
 GCP_SERVICE_USAGE_URL = "https://serviceusage.googleapis.com/v1/projects/"
 
 
 async def get_all_disabled_apis(context: MetricsContext, token: str, project_id: str):
-    url = f"{GCP_SERVICE_USAGE_URL}{project_id}/services?filter=state:DISABLED"
+    base_url = f"{GCP_SERVICE_USAGE_URL}{project_id}/services?filter=state:DISABLED"
     headers = {"Authorization": "Bearer {token}".format(token=token)}
     disabled_apis = set()
     try:
-        response = await context.gcp_session.get(url, headers=headers)
-        if response.status != 200:
-            context.log(f'Http error: {response.status}, url: {response.url}, reason: {response.reason}')
-            return set()
+        response = await context.gcp_session.get(base_url, headers=headers, raise_for_status=True)
         disabled_services_json = await response.json()
         disabled_services = disabled_services_json.get("services", [])
         disabled_apis.update({disable_service.get("config", {}).get("name", "") for disable_service in disabled_services})
         while disabled_services_json.get("nextPageToken"):
```

Comment:
I don't understand it. You say that for non-200 responses the code looped forever. Was it on this while-loop? Because if there's non-200 response we shouldn't even have reached this loop (we would've returned empty set in line 28).

Reply:
That is true, and if the first call fails, we don't get disabled APIs, continue and everything is fine.
However, oftentimes the initial call succeeds, yet most of the following ones do not leading to the URL becoming too long and eventually no requests passing at all and the loop becoming infinite. 
One approach to consider could be only changing the URL param appending, but I'm not sure if there is much upside - if we sometimes fail because of a temporary issue, we'll just do some more requests than needed later to the disabled APIs.

Reply:
That is true, and if the first call fails, we don't get disabled APIs, continue and everything is fine.
However, oftentimes the initial call succeeds, yet most of the following ones do not leading to the URL becoming too long and eventually no requests passing at all and the loop becoming infinite.
One approach to consider could be only changing the URL param appending, but I'm not sure if there is much upside - if we sometimes fail because of a temporary issue, we'll just do some more requests than needed later to the disabled APIs.

Reply:
Ahh, I see, because `disabled_services_json` is only updated when we get 200 in the loop. Otherwise we reuse the value that we received in the initial request all the time